### ISSUE TITLE
[Sys2x6 Feature Request] Allow JVS ID string to be configured through arcadedefs

### ISSUE THREAD

**Issue by Hydreigon223** [jpd002/Play-#1254]
https://github.com/jpd002/Play-/blob/namco_system_256/Source/iop/Iop_NamcoArcade.cpp#LL156C3-L170C4
`case JVS_CMD_IOIDENT:
		{
			(*output++) = 0x01; //Command success
			(*dstSize)++;

			//const char* boardName = "namco ltd.;RAYS PCB;";
			const char* boardName = "namco ltd.;TSS-I/O;";
			size_t length = strlen(boardName);

			for(int i = 0; i < length + 1; i++)
			{
				(*output++) = boardName[i];
				(*dstSize)++;
			}
		}`

We should have the string "boardname" determined through arcadedefs instead of a hardcoded default value. Keeping the ID string as TSS-I/O  is not very sane as it's a one player optical lightgun board used in Time Crisis 2 and 3. The Taito games running on System 2x6 (at least Battle Gear 3) are known for checking this string for a specific value (as well as checking the I/O board serial port for a servo motor board but that's a different topic).

**Comment by jpd002** [comment]
Yup, makes sense, definitely an improvement I need to make. That ID seemed to work for games I've tested, but I haven't tested most of the games yet.

**Comment by Keropon** [comment]
Does it just compare the ID string? I know each IO has different configurations on available channels and stuff like that

**Comment by Hydreigon223** [comment]
> Does it just compare the ID string? I know each IO has different configurations on available channels and stuff like that

For BG3, it's stated on s2x6 compatibility document that it requires a very specific JVS IO board (likely checks the ID along with I/O bits and A/D channels being an equal number). I also made the mistake of mentioning where the JVS serial port connects to and correctly edited that it connects to a card reader instead. The servo motor board on the other hand connects through the same serial port used to connect an STR motor feedback pcb.

Idk if any other checks are used for the card reader but that in itself is nothing fancy. All it just does is read a serial ID. The profile saving/loading on the other hand is done with an online server instead (requires usb ethernet adapter support and later nesys server emulation with the latter being "someone else's" responsibility when that time comes).

**Comment by Keropon** [comment]
Do you know if BG3 has a hidden offline mode as im@s does? (this is obviously not related to the issue)

Should the correct approach be an enum where Play has a list of general usage board names or should it be so it reads just the string?. I mean, I don't know if bloating the .arcadedef with each setting for the arcade games is the idea or not, it should depend on @jpd002 imo. I could try to help implementing it

**Comment by jpd002** [comment]
Yeah, I think it makes sense to add an enum for the board type with a way to "override" the default in `arcadedef` files. PR is welcome.

**Comment by jpd002** [comment]
`boardId` string value was added in `arcadedef` in #1516 .
